### PR TITLE
update fsnotify to v1.6.0

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
https://github.com/fsnotify/fsnotify/blob/main/CHANGELOG.md#160---2022-10-13

#### Which issue(s) this PR fixes:
xref #112574

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by dims** [comment]
/approve
/lgtm

thanks @pacoxu looks like there's a bunch of things that got fixed.

File: vendor/github.com/fsnotify/fsnotify/fsnotify.go [hunk 1]
```diff
@@ -32,38 +40,42 @@ const (
...
+// Has reports if this operation has the given operation.
+func (o Op) Has(h Op) bool { return o&h == h }
+
+// Has reports if this event has the given operation.
+func (e Event) Has(op Op) bool { return e.Op.Has(op) }
+
+// String returns a string representation of the event with their path.
 func (e Event) String() string {
-	return fmt.Sprintf("%q: %s", e.Name, e.Op.String())
+	return fmt.Sprintf("%-13s %q", e.Op.String(), e.Name)
```
[the remark this answers is not in the document]
Comment (pacoxu) [reply vendor/github.com/fsnotify/fsnotify/fsnotify.go:80]: Not used if I search the code of k/k correctly.
See https://github.com/kubernetes/kubernetes/issues/112574#issuecomment-1277206197

[the remark this answers is not in the document]
Comment (BenTheElder) [reply vendor/github.com/fsnotify/fsnotify/fsnotify.go:80]: I don't think so either but searching for String() usage is hard, can be indirect.
Not seriously worried about it, just noting this is a spot I'd be coming back to if issues cropped up.

**Pull request merged**